### ISSUE TITLE
Int8DynamicActivationInt8WeightConfig noticeably slower than torch.ao's quantize_dynamic()

### ISSUE THREAD

**Issue by mscdex** [pytorch/ao#4365]
### Description

This isn't so much a regression within torchao itself as it is a regression compared to torch.ao, but when I run for example:

```
pocket-tts generate --voice voice.wav --language english_2026-04 --quantize --device cpu --text "hello world, this is a very long sentence so that the inference will take long enough for us to do some chunking and show that the same performance is repeated. hello world, this is a very long sentence so that the inference will take long enough for us to do some chunking and show that the same performance is repeated."
```

(where voice.wav is simply [this wav](https://huggingface.co/onnx-community/chatterbox-multilingual-ONNX/blob/2301da1364e51845d60726654bcd0a85d0e96f02/default_voice.wav) converted to a 16-bit signed integer wav)

With torchao installed (which is automatically detected and used), I get ~5.33x faster than realtime performance. If I don't install or I uninstall torchao so that the fallback to torch.ao is used, I get ~7.17x faster than realtime.

I'm not familiar enough with pytorch, torchao, or even Python for that matter to know where the fault lies.

I did try various things like adding a `module.compile()` after the `quantize_()` call, but that actually made things significantly slower (I guess because there is a lot of recompiling going on or something similar).

I also tried reducing the quantized parts of each layer to see if it was just the attention or just the ffn quantizations that were the problem and removing either did not result in equal or better performance for torchao.

### Before/After numbers

```text
With torchao: ~5.33x faster than realtime
With torch.ao: ~7.17x faster than realtime
```

### Model and quantization config

[pocket-tts](https://github.com/kyutai-labs/pocket-tts) 2.1.0

[Here](https://github.com/kyutai-labs/pocket-tts/blob/670e7772607d182929594fdf11540d7565246477/pocket_tts/quantization.py) is the quantization code. As far as I can tell it's using the fbgemm backend to apply Int8DynamicActivationInt8WeightConfig quantization to the attention and ffn parts of flow lm transformer layers.

### Environment

```text
Python 3.12.3 + CPU (tested with: Ryzen 7700X, Xeon Gold 6238)
torch==2.11.0+cpu
torchao==0.17.0
torchaudio==2.11.0+cpu
```

**Comment by Xia-Weiwen** [comment]
Hi @mscdex Please try the following:
```python
torch._inductor.config.freezing = True

m = Model().eval()
quantize_(m, Int8DynamicActivationInt8WeightConfig())
mc = torch.compile(m)
# then warmup and benchmarks
```

**Comment by mscdex** [comment]
> Please try the following:

With those changes I get even slower performance: ~1.11x after warmup

I'm not sure if it's relevant or not, but I did notice a lot of `AffineQuantizedTensor`, `PlainAQTTensorImpl`, and `PlainLayout` deprecation warnings printed to the console while the process ran.

Additionally, I saw a recompilation limit warning. While increasing the limit to something higher (I used 64) got rid of the warning, it did not have any measurable effect on the slower performance.

```
W0506 08:36:13.713000 3366316 torch/_dynamo/convert_frame.py:1743] [1/8] torch._dynamo hit config.recompile_limit (8)
W0506 08:36:13.713000 3366316 torch/_dynamo/convert_frame.py:1743] [1/8]    function: 'forward' (/tmp/pocket-tts-v2-torch/venv/lib/python3.12/site-packages/pocket_tts/modules/transformer.py:135)
W0506 08:36:13.713000 3366316 torch/_dynamo/convert_frame.py:1743] [1/8]    last reason: 1/7: ___check_obj_id(self._modules['in_proj']._parameters['weight'], 126720150387552), type=LinearActivationQuantizedTensor(AffineQuantizedTensor(tensor_impl=PlainAQTTensorImpl(data=tensor([[  41,   50,   21,  ...,  -18,  -13,  -11],
W0506 08:36:13.713000 3366316 torch/_dynamo/convert_frame.py:1743] [1/8]         [  12,   27,  -15,  ...,   29,    1,    3],
W0506 08:36:13.713000 3366316 torch/_dynamo/convert_frame.py:1743] [1/8]         [  23,   80,    0,  ...,    2,   36,    9],
W0506 08:36:13.713000 3366316 torch/_dynamo/convert_frame.py:1743] [1/8]         ...,
W0506 08:36:13.713000 3366316 torch/_dynamo/convert_frame.py:1743] [1/8]         [  -3,    3,   -6,  ...,    0,   10,   -2],
W0506 08:36:13.713000 3366316 torch/_dynamo/convert_frame.py:1743] [1/8]         [-120,   -2,   -1,  ...,   -3,   -4,    6],
W0506 08:36:13.713000 3366316 torch/_dynamo/convert_frame.py:1743] [1/8]         [ -14,  -18,    5,  ...,    1,   12,   -3]], dtype=torch.int8)... , scale=tensor([0.0006, 0.0005, 0.0005,  ..., 0.0004, 0.0004, 0.0005])... , zero_point=None... , _layout=PlainLayout()), block_size=(1, 1024), shape=torch.Size([3072, 1024]), device=cpu, dtype=torch.float32, requires_grad=False), <function _int8_symm_per_token_reduced_range_quant at 0x734056028cc0>, quant_kwargs={}))  # return F.linear(input, self.weight, self.bias)  # nn/modules/linear.py:134 in forward
W0506 08:36:13.713000 3366316 torch/_dynamo/convert_frame.py:1743] [1/8] User stack trace:
W0506 08:36:13.713000 3366316 torch/_dynamo/convert_frame.py:1743] [1/8]   File "/tmp/pocket-tts-v2-torch/venv/lib/python3.12/site-packages/pocket_tts/modules/transformer.py", line 138, in forward
W0506 08:36:13.713000 3366316 torch/_dynamo/convert_frame.py:1743] [1/8]     projected = self.in_proj(query)
W0506 08:36:13.713000 3366316 torch/_dynamo/convert_frame.py:1743] [1/8]   File "/tmp/pocket-tts-v2-torch/venv/lib/python3.12/site-packages/torch/nn/modules/linear.py", line 134, in forward
W0506 08:36:13.713000 3366316 torch/_dynamo/convert_frame.py:1743] [1/8]     return F.linear(input, self.weight, self.bias)
```

**Comment by Xia-Weiwen** [comment]
Can you share a reproducer?

**Comment by mscdex** [comment]
> Can you share a reproducer?

I'm not sure what you're looking for beyond a `pip install pocket-tts` followed by the command line in the OP. Then `pip install torchao` and re-running the command.

**Comment by Xia-Weiwen** [comment]
It does not show how it uses torchao
